### PR TITLE
Add Nuget Pipeline (#696)

### PR DESCRIPTION
**Pull request by xXLAOKOONXx**
Implementation of https://github.com/BenFradet/RiotSharp/issues/696

Checkout PIPELINES.md for some explaination on usage.

### CODE REVIEW

File: .github/workflows/nuget_action.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      run: dotnet pack -p:Configuration=Release -p:Version=${{ steps.get_version.outputs.VERSION }} --output ../npkgs
+    - name: Check npkg folder
+      working-directory: ./npkgs
+      run: ls
+    - name: Publish Nuget on nuget.org
+      working-directory: ./npkgs
+      run: dotnet nuget push RiotSharp.${{ steps.get_version.outputs.VERSION }}.nupkg -s https://nuget.org -k ${{secrets.NUGET_API_KEY}}
+    - name: Publish Nuget on github
+      working-directory: ./npkgs
+      run: dotnet nuget push RiotSharp.${{ steps.get_version.outputs.VERSION }}.nupkg -s https://nuget.pkg.github.com/BenFradet/index.json -k ${{secrets.MY_GITHUB_KEY}}
```
Comment (BenFradet) [inline .github/workflows/nuget_action.yml]: I'm assuming this should be a github api token, do you know what scopes it needs?

File: PIPELINES.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+## Repository Requirements
+In order for the pipeline to run there need to be some secrets in place.  
+Secrets can be added in the repository -> Settings -> Secrets -> (top right) New repository secret  
+Secrets required for the pipeline:  
+- NUGET_API_KEY being an api key for nuget.org [nuget api key](https://www.nuget.org/account/apikeys)
+
+## Usage
+Create a Github Release with a tag in the format of versioning (eg 5.0.0 or 5.0.0-alpha).  
+Checkout the Actions tab to figure out whether the pipe run smoothly or errors occured.
```
Comment (BenFradet) [inline PIPELINES.md]:
```suggestion
Checkout the Actions tab to figure out whether the pipe ran smoothly or errors occurred.  
```

File: .github/workflows/nuget_action.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      run: dotnet restore
+    - name: Build    
+      working-directory: ./RiotSharp
+      run: dotnet build --no-restore      
+    - name: Build Nuget
+      working-directory: ./RiotSharp
+      run: dotnet pack -p:Configuration=Release -p:Version=${{ steps.get_version.outputs.VERSION }} --output ../npkgs
+    - name: Check npkg folder
+      working-directory: ./npkgs
+      run: ls
```
Comment (BenFradet) [inline .github/workflows/nuget_action.yml]: can we remove this?

Comment (xXLAOKOONXx) [reply .github/workflows/nuget_action.yml]: Yes

File: .github/workflows/nuget_action.yml (hunk 1 above)
Comment (xXLAOKOONXx) [reply .github/workflows/nuget_action.yml]: NUGET_API_KEY is described in markdown file.  
`MY_GITHUB_KEY` is outdated, GITHUB_TOKEN should do the job and this gets shipped with every action with the needed acces by default (uploading packages)

Comment (xXLAOKOONXx) [reply .github/workflows/nuget_action.yml]: Added an explaination in markdown regarding GITHUB_TOKEN

**Review by BenFradet — APPROVED** [review]

**Comment by BenFradet** [comment]
Thanks!

**Pull request merged**